### PR TITLE
add compile time settings into version output

### PR DESCRIPTION
**Pull request by cgzones**
example output:

$ logrotate --version
logrotate 3.12.3

    Default mail command: /bin/mail
    Compress command:     /bin/gzip
    Uncompress command:   /bin/gunzip
    Compress extension:   .gz
    State file path:      /var/lib/logrotate.status
    ACL support:          yes
    SELinux support:      yes

### CODE REVIEW

**Comment by glensc** [comment]
what has always bothered me, that tools use `stderr` for intended messages (not errors), so they are not easily pipeable. imho version output should go to `stdout`.

File: logrotate.c [hunk 1]
```diff
@@ -2600,6 +2600,22 @@ int main(int argc, const char **argv)
 		break;
 	case 'V':
 	    fprintf(stderr, "logrotate %s\n", VERSION);
+	    fprintf(stderr, "\n");
+	    fprintf(stderr, "    Default mail command: %s\n", DEFAULT_MAIL_COMMAND);
+	    fprintf(stderr, "    Compress command:     %s\n", COMPRESS_COMMAND);
+	    fprintf(stderr, "    Uncompress command:   %s\n", UNCOMPRESS_COMMAND);
+	    fprintf(stderr, "    Compress extension:   %s\n", COMPRESS_EXT);
+	    fprintf(stderr, "    State file path:      %s\n", STATEFILE);
```
Comment (kdudka) [inline logrotate.c]: I think we should use the ```Default ``` prefix in all the lines above.  This is especially important for the state file, which can be overridden by a command-line option (and certain distros use the option in default cron jobs to override logrotate's default) but I would put it everywhere to stay consistent.

**Comment by kdudka** [comment]
@glensc Agreed.  Are you proposing to change the output stream for ```--version``` along with this PR?

In any case, such a change needs to be properly advertised in release notes because it is a backward incompatible change.

**Comment by glensc** [comment]
@kdudka yes, change in this PR, as depending how `--version` output is parsed by external scripts, it could break scripts anyway.

for example `logrotate --version 2>&1 | awk '{print $2}'`.

**Comment by kdudka** [comment]
@cgzones Are you fine with including the change of output stream in this PR?

**Comment by kdudka** [comment]
Can I abbreviate ```fprintf(stdout, ...)``` as ```printf(...)```?

You do not need to resbumit it, of course...

**Comment by cgzones** [comment]
> Can I abbreviate fprintf(stdout, ...) as printf(...)?

yes, of course

**Comment by kdudka** [comment]
OK, pushing now... Thanks!

**Pull request closed without merging**